### DISCUSSION TITLE
200Something went wrong... Code: 403 with qBittorrent WebUI

### DISCUSSION THREAD

**Discussion by steakhutzeee** [linuxserver/Heimdall#745] [App issues]
Hi :)

On my Odroid HC2 running OMV5 on top of Armbian i set up heimdall with this container https://hub.docker.com/r/linuxserver/heimdall

I'm also using qBittorrent with this container https://hub.docker.com/r/linuxserver/qbittorrent

I can reach the qBittorrent container just fine via Heimdall but when i try to put in my Web UI credentials in Heimdall to see the Enhanced App i receive the following pop up error:

"200Something went wrong... Code: 403"

Any idea why?

Thanks!

**Comment by bn326160** [comment]
Wanted to pitch in to mention I have the same issue on Docker on Ubuntu Server LTS 20.04.

However, when I added it anyway, it seems to be working?

**Comment by steakhutzeee** [comment]
Yes! It actually works, even with that error. That was not working when i opened the issue. Strange.

**Comment by jmu** [comment]
I googled and found the solution by reddit user u/joseamaria
Web UI Tab > Enable Cross-Site Request Forgery (CSRF) protection

https://www.reddit.com/r/docker/comments/gfkwwq/qbittorrent_webui_linking_problem/

**Comment by steakhutzeee** [comment]
> 
> 
> I googled and found the solution by reddit user u/joseamaria
> Web UI Tab > Enable Cross-Site Request Forgery (CSRF) protection
> 
> https://www.reddit.com/r/docker/comments/gfkwwq/qbittorrent_webui_linking_problem/

Hi,
the issue posted by the reddit user is different by mine.

Anyway i have that option disabled by long time and the issue is still present.

**Comment by steakhutzeee** [comment]
Actually with latest version of qbittorrent and heimdall the issue is present again, nothing happens when using qbittorrent credentials.

**Comment by ahmedelakkad** [comment]
r

> Actually with latest version of qbittorrent and heimdall the issue is present again, nothing happens when using qbittorrent credentials.

Same!

**Comment by avdept** [comment]
Any update on this?

**Comment by dexogen** [comment]
Well, I found workaround.
1. Connect qBittorrent container to the same docker network (e.g. heimdall_default).
```
$ docker inspect qbittorrent
...
"Networks": {
                "heimdall_default": {
                    "IPAMConfig": {},
                    "Links": null,
                    "Aliases": [
                        "7aca1bf06489",
                        "qbittorrent"
...
$ docker inspect heimdall
...
"Networks": {
                "heimdall_default": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "heimdall",
                        "5a059215554e"
                    ]
...
```
2. Add this subnet into bypass whitelist (Options > WebUI > Authentication > Bypass authentication for clients)
![image](https://user-images.githubusercontent.com/2457575/173203069-b680144e-2108-450d-9432-31e2a61469d5.png)
3. Go to the Heimdall and insert URL the using internal ip or alias w/o username and password. If "Host Header Check" is enabled, then add an alias to the server's domains.
![image](https://user-images.githubusercontent.com/2457575/173203081-25bfd131-5e2e-4e2c-a26a-9951c2bd4520.png)
4. Success
![image](https://user-images.githubusercontent.com/2457575/173203097-45c873dc-d173-4cd9-8afa-21029922edf5.png)

**Reply by ProfessorX** [reply]
this one works for me on unraid.

**Reply by ChristophorusReyhan** [reply]
Works, but insecure 🤕

**Comment by Hurray0** [comment]
Any update?
I cannot use `subnet bypass whitelist` which is not safety for me.
Wish there can be other method to solving this problem

**Comment by ChristophorusReyhan** [comment]
Found it. Check your qbittorrent logs (for me /.local/share/qBittorrent/logs/qbittorrent.log). 
If you got 
`(W) 2024-10-26T14:47:19 - WebAPI login failure. Reason: IP has been banned, IP: ::ffff:10.0.0.104, username: admin`
Then try to reboot your qbittorrent service (for me `systemctl restart qbittorrent-nox.service`).

Test it again.
